### PR TITLE
Exposed all ticket options (master)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -140,12 +140,18 @@ This endpoint provides a service for the generation of a read-only iRODS ticket
 
 **Parameters:**
 - path: The url encoded logical path to a collection or data object for which access is desired
+- type: The type of ticket to create. The value must be either read or write. Defaults to read
 - use_count: The maximum number of times the ticket can be used. Defaults to 0 (unlimited use)
-- seconds_until_expiration: The number of seconds before the ticket will expire. Defaults to 30 seconds
+- write_file_count: The maximum number of writes allowed to a data object. Defaults to 0 (unlimited writes)
+- write_byte_count: The maximum number of bytes allowed to be written to data object. Defaults to 0 (unlimited bytes)
+- seconds_until_expiration: The number of seconds before the ticket will expire. Defaults to 0 seconds
```
Comment (trel) [inline README.md:147]: add " (no expiration)"

Comment (korydraughn) [reply README.md]: done

File: irods_specific_implementation/irods_rest_access_implementation.h [hunk 2]
```diff
@@ -135,41 +150,134 @@ namespace irods::rest
...
             const auto end_time = fmt::format("{}", duration_cast<seconds>(tp.time_since_epoch()).count());
+#else
+            using namespace boost::posix_time;
+
+            const auto secs = std::stoll(_seconds_until_expiration.getOrElse("0"));
+            debug("seconds_until_expiration = [{}]", secs);
+
+            const auto expiration_timestamp = second_clock::universal_time() + seconds{secs};
+            const auto end_time = fmt::format("{:011}", to_time_t(expiration_timestamp));
+#endif
```
[the remark this answers is not in the document]
Comment (korydraughn) [reply irods_specific_implementation/irods_rest_access_implementation.h]: decided to go with std::chrono implementation. the system_clock type is normally implemented using UTC time.

**Comment by korydraughn** [comment]
i've disabled the get/put remote configuration endpoints until support in the server is fleshed out.

i also updated the ticket endpoint to delete the ticket if an exception is thrown at any point while handling the request.

**Comment by korydraughn** [comment]
if there is nothing else we want to change for this PR, i'll add the #'s

File: README.md [hunk 3]
```diff
@@ -140,12 +140,18 @@ This endpoint provides a service for the generation of a read-only iRODS ticket
...
+- write_byte_count: The maximum number of bytes allowed to be written to data object. Defaults to 0 (unlimited bytes)
+- seconds_until_expiration: The number of seconds before the ticket will expire. Defaults to 0 (no expiration)
+- users: A comma-separated list of iRODS users who are allowed to use the generated ticket
+- groups: A comma-separated list of iRODS groups that are allowed to use the generated ticket
+- hosts: A comma-separated list of hosts that are allowed to use the ticket
 
 **Example CURL Command:**
 ```
-curl -X POST -H "Authorization: ${TOKEN}" 'http://localhost/irods-rest/0.8.0/access?path=%2FtempZone%2Fhome%2Frods%2Ffile0'
+curl -X POST -H "Authorization: ${TOKEN}" 'http://localhost/irods-rest/0.8.0/access?path=%2FtempZone%2Fhome%2Frods%2Ffile0&type=write&write_file_count=10'
```
Comment (trel) [inline README.md:154]: let's change this to 0.9.0 for the next release...

File: irods_specific_implementation/irods_rest_access_implementation.h [hunk 4]
```diff
@@ -135,41 +167,137 @@ namespace irods::rest
...
+            if ("0" == secs_until_exp_string) {
+                return;
+            }
+
             trace("Setting expiration on ticket [{}] ...", _ticket_id);
 
+            const auto secs = std::stoll(secs_until_exp_string);
+
+            if (secs < 0) {
+                THROW(SYS_INVALID_INPUT_PARAM, "[seconds_until_expiration] must be greater or equal to zero.");
```
Comment (trel) [inline irods_specific_implementation/irods_rest_access_implementation.h:217]: "greater than"

File: README.md (hunk 3 above)
Comment (korydraughn) [reply README.md:154]: sure. i'll add another commit that updates everything to 0.9.0

File: packaging/test_irods_client_rest_cpp.py [hunk 5]
```diff
@@ -42,24 +42,117 @@ def test_access(self):
...
+                json_string = irods_rest.access(token, logical_path, _seconds_until_expiration=-1)
+                assert(json_string.find('error') > 0)
+                json_object = json.loads(json_string)
+                self.assertEqual(json_object['error_code'], -130000)
+
+            finally:
+                os.remove(file_name)
+                admin.assert_icommand(['irm', '-f', file_name])
+
+    @unittest.skip('Disabled until support in iRODS server is fleshed out.')
```
Comment (trel) [inline packaging/test_irods_client_rest_cpp.py:149]: let's say "is complete"

File: packaging/test_irods_client_rest_cpp.py [hunk 6]
```diff
@@ -42,24 +42,117 @@ def test_access(self):
...
     def test_get_configuration(self):
         token  = irods_rest.authenticate('rods', 'rods', 'native')
         result = irods_rest.get_configuration(token)
         assert(result.find('advanced_settings') != -1)
 
+    @unittest.skip('Disabled until support in iRODS server is fleshed out.')
```
Comment (trel) [inline packaging/test_irods_client_rest_cpp.py:155]: let's say "is complete"

File: README.md (hunk 3 above)
Comment (korydraughn) [reply README.md]: done.

should i create a new issue for bumping the version number? or should i use the issue for "Release Activities"?

File: irods_specific_implementation/irods_rest_access_implementation.h (hunk 4 above)
Comment (korydraughn) [reply irods_specific_implementation/irods_rest_access_implementation.h]: done

File: packaging/test_irods_client_rest_cpp.py (hunk 5 above)
Comment (korydraughn) [reply packaging/test_irods_client_rest_cpp.py]: done

File: packaging/test_irods_client_rest_cpp.py (hunk 6 above)
Comment (korydraughn) [reply packaging/test_irods_client_rest_cpp.py]: done

File: README.md (hunk 3 above)
Comment (korydraughn) [reply README.md]: nevermind. im going to create a new issue for bumping the version.

**Comment by korydraughn** [comment]
added #

**Pull request merged**